flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java

```diff
@@ -1589,6 +1601,15 @@ private <T> T execute(Supplier<T> supplier) {
                 }
             }
         }
+
+        @Override
+        public boolean hasChanges() throws IllegalStateException {
+            if (this.binder == null) {
+                throw new IllegalStateException();
```

Comment:
We need an informative message for this exception

```suggestion
                throw new IllegalStateException("This Binding is no longer attached to a Binder");
```